### PR TITLE
Feat: Swift Testing framework (2/3)

### PR DESCRIPTION
**Pull request by DanielCech**
- The dependency injection tests are converted to the new Testing library
- Tests are adapted to the parameters packs approach. There is also one test for checking the maximum allowed argument count.
- I removed the default scope functionality because it caused some ambiguity in parameter type inference. We need to be specific about the scope, which is, imho, a good thing.
- I tested this branch (including multiple argument propagation) on TechnoAlpin in the `dc/feat/new-di-test` branch

This PR is built on top of https://github.com/strvcom/ios-dependency-injection/pull/22

### CODE REVIEW

**Review by cejanen — APPROVED** [review]
With Testing would be nice if we utilize features to better structure tests (suite, test domain etc)

File: Tests/Container/Async/AsyncBaseTests.swift [hunk 1]
```diff
@@ -6,80 +6,111 @@
...
-        XCTAssertTrue(resolvedDependency1 === resolvedDependency2, "Container returned different instance")
-    }
-
-    func testDependencyRegisteredInDefaultScopeWithExplicitType() async {
-        await container.register(type: SimpleDependency.self, in: .shared) { _ -> SimpleDependency in
+import Testing
+
+@Suite("Container/Async/Base Registration", .tags(.async, .base))
+struct AsyncBaseTests {
+//    @Test("Dependency registered in default scope")
```
Comment (TParizek) [inline Tests/Container/Async/AsyncBaseTests.swift]: Is there a reason for having this test commented out?

**Review by TParizek — COMMENTED** [review]
This PR looks great now 👍 Please just remove the one commented out test case and let's wait for CI to be available, so we can check changes - it should be available very soon.

File: Tests/Container/Sync/BaseTests.swift [hunk 2]
```diff
@@ -6,106 +6,153 @@
...
+        let resolvedDependency2: SimpleDependency = subject.resolve()
 
-        XCTAssertTrue(resolvedDependency1 === resolvedDependency2, "Container returned different instance")
+        // Then
+        #expect(resolvedDependency1 === resolvedDependency2)
     }
 
-    func testSharedDependency() {
-        container.register(in: .shared) { _ -> SimpleDependency in
+//    @Test("Dependency registered in default scope")
```
[the remark this answers is not in the document]
Comment (DanielCech) [reply Tests/Container/Sync/BaseTests.swift]: Sure, fixed.

**Review by TParizek — DISMISSED** [review]

**Review by TParizek — APPROVED** [review]
